src/test/java/com/cdancy/jenkins/rest/features/JobsApiMockTest.java

```diff
@@ -577,6 +578,37 @@ public void testGetParamsWhenNoBuildParams() throws Exception {
         }
     }
 
+    public void testGetParamsWhenEmptyorNullParams() throws Exception {
+        MockWebServer server = mockWebServer();
+
+        String body = payloadFromResource("/build-info-empty-and-null-params.json");
+        server.enqueue(new MockResponse().setBody(body).setResponseCode(200));
+        JenkinsApi jenkinsApi = api(server.getUrl("/"));
+        JobsApi api = jenkinsApi.jobsApi();
+        try {
+            List<Action> output = api.buildInfo(null,"fish", 10).actions();
+            List<Parameter> parameters = new ArrayList<>();
+            for (Action action : output) {
+                if (!action.parameters().isEmpty()) {
+                    parameters = action.parameters();
+                }
+            }
+            for (Parameter parameter : parameters) {
```

Comment:
I intentionally added the test like this to provide an alternative to get(0) etc for getting params. I can change it if suggested otherwise.

Reply:
Well, thanks for the tip, but I don't think it is necessary. In fact it makes the test slightly harder to read.

Reply:
Agreed. Makes it harder to read. Lets just use what comes OOTB and iterate over that.